packages/frontend2/src/app/(legacy)/scaling-next/projects/[slug]/tvl-breakdown/_components/tables/canonically-bridged-table.tsx

```diff
@@ -0,0 +1,65 @@
+'use client'
+
+import {
+  type ExpandedState,
+  getCoreRowModel,
+  getExpandedRowModel,
+} from '@tanstack/react-table'
+import { useState } from 'react'
+import { TokenTable } from '~/app/_components/table/token-breakdown-table'
+import { useTable } from '~/hooks/use-table'
+import { type ExtendedProjectTvlBreakdown } from '../../_utils/assign-token-meta-to-breakdown'
+import { canonicallyBridgedColumns } from './columns/canonically-bridged-columns'
+import { TableSum } from './table-sum'
+
+export type CanonicallyBridgedTokenEntry =
+  ExtendedProjectTvlBreakdown['breakdown']['canonical'][number]
+
+interface Props {
+  tokens: CanonicallyBridgedTokenEntry[]
+}
+
+export function CanonicallyBridgedTable(props: Props) {
+  const sum = props.tokens.reduce((acc, token) => {
+    return acc + Number(token.usdValue)
+  }, 0)
+
+  const [expanded, setExpanded] = useState<ExpandedState>({})
+
+  const table = useTable({
+    data: props.tokens,
+    columns: canonicallyBridgedColumns,
+    state: {
+      expanded,
+    },
+    initialState: {},
+    onExpandedChange: setExpanded,
+    getSubRows: (row) =>
+      row.escrows.length > 1
+        ? row.escrows.map((escrow) => ({
+            ...row,
+            // Port escrow to sub row
+            amount: escrow.amount,
+            usdValue: escrow.usdValue,
+            escrows: [escrow],
+          }))
+        : undefined,
```

Comment:
why doing it this way instead of grouping?

Reply:
expandable rows are way easier to work with - the table doesn't need any more changes and handling - unless you have an idea on how to do it properly - then we can discuss it